julia-package/brainflow/src/board_shim.jl

```diff
@@ -49,12 +49,35 @@ end
     file::String = ""
 end
 
+struct BoardShim
+
+    master_board_id::Int32
+    board_id::Int32
+    input_json::String
+
+    function BoardShim(id::Integer, params::BrainFlowInputParams)
+        master_id = id
+        if id == Integer(STREAMING_BOARD) || id == Integer(PLAYBACK_FILE_BOARD)
+            try
+                master_id = parse(Int, params.other_info)
+            catch
+                throw(BrainFlowError("you need to provide master board id to other_info field of BrainFlowInputParams", Integer(INVALID_ARGUMENTS_ERROR)))
+            end
+        end
+        new(master_id, id, JSON.json(params))
+    end
+
+end
+BoardShim(id::BoardIds, params::BrainFlowInputParams) = BoardShim(Integer(id), params)
+BoardShim(id) = BoardShim(id, BrainFlowInputParams())
+
 @brainflow_rethrow function get_sampling_rate(board_id::BoardIdType)
     val = Vector{Cint}(undef, 1)
     ccall((:get_sampling_rate, BOARD_CONTROLLER_INTERFACE), Cint, (Cint, Ptr{Cint}), Int32(board_id), val)
     value = val[1]
     return value
 end
+get_sampling_rate(board_shim::BoardShim) = get_sampling_rate(board_shim.master_board_id)
```

Comment:
for the same reason as above lets remove these overloaded methods, I understand the motivation for it, but the rule is to keep the same API between languages

Reply:
Ok, though I hate writing more code in my scripts